File: backend/src/main/java/com/campus/Campus_Connect/features/event/dto/request/CreateEventRequest.java
package com.campus.Campus_Connect.features.event.dto.request;


import com.campus.Campus_Connect.features.event.entity.enums.VisibilityType;
import com.campus.Campus_Connect.features.event.entity.enums.RegistrationType;
import jakarta.validation.constraints.*;
import lombok.*;
import java.time.Instant;




@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateEventRequest {

    @NotBlank(message = "Title is required.")
    private String title;

    private String description;

    @NotNull(message = "Latitude is required.")
    @DecimalMin(value = "-90.0", message = "Latitude must be at least -90.")
    @DecimalMax(value = "90", message = "Latitude must be at most 90.")
    private Double latitude;

    @NotNull(message = "Longitude is required.")
    @DecimalMin(value = "-180.0", message = "Longitude must be at least -180.")
    @DecimalMax(value = "180.0", message = "Longitude must be at most 180.")
    private Double longitude;

    @NotNull(message = "Start time is required.")
    private Instant startTime;
    private Instant endTime;

    private Integer clubId;

    private String hostName;

    private String venue;

    @NotBlank(message = "Visibility type is required.")
    private VisibilityType visibilityType;

    private String visibilityValue;

    @NotBlank(message = "Registration type is required.")
    private RegistrationType registrationType;

    private String registrationLink;

    @NotNull(message = "Priority is required.")
    @Min(value = 1, message = "Priority must be between 1 and 6.")
    @Max(value = 6, message = "Priority must be between 1 and 6.")
    private Integer priority;
}

File: android-app/app/src/main/java/com/example/campusconnect/feature/events/data/remote/request/CreateEventRequest.kt
package com.example.campusconnect.feature.events.data.remote.request

data class CreateEventRequest(

    val title: String,

    val description: String? = null,

    val latitude: Double,

    val longitude: Double,

    val startTime: String,

    val endTime: String? = null,

    val clubId: Int? = null,

    val hostName: String? = null,

    val venue: String? = null,

    val visibilityType: String,

    val visibilityValue: String? = null,

    val registrationType: String,

    val registrationLink: String? = null,

    val priority: Int,

    val categoryId: Int
)